sklearnex/decomposition/pca.py

```diff
@@ -210,6 +210,24 @@ def fit_transform(self, X, y=None):
                 # Scikit-learn PCA["covariance_eigh"] was fit
                 return self._transform(X_fit, xp, x_is_centered=x_is_centered)
 
+        @wrap_output_data
+        def inverse_transform(X):
+            xp, _ = get_namespace(X)

```

Comment:
weakness in input checking matches issues in sklearn, currently a PR to fix it is in: scikit-learn/scikit-learn#29310